### PR TITLE
INT-40 fix SubmitReview

### PR DESCRIPTION
**Pull request by rkaczka-vd**

### CODE REVIEW

File: IndicoV2/Extensions/Jobs/JobAwaiter.cs [hunk 1]
```diff
@@ -27,7 +27,8 @@ public async Task<JToken> WaitReadyAsync(string jobId, TimeSpan checkInterval, C
 
             if (status != JobStatus.SUCCESS)
             {
-                throw new JobNotSuccessfulException(status);
+                var failReason = await _jobsClient.GetFailReasonAsync(jobId);
```
Comment (goatrocks) [inline IndicoV2/Extensions/Jobs/JobAwaiter.cs:30]: This logic overall needs some refinement, given there are multiple statuses like PENDING_AUTOREVIEW, or if perhaps someone calls it on a doc still in PROCESSING. We don't want to fail on that.

**Review by goatrocks — COMMENTED** [review]
Couple comments but nothing major!

**Review by szymon-socha — APPROVED** [review]

File: IndicoV2/Extensions/Jobs/JobAwaiter.cs (hunk 1 above)
Comment (rkaczka-vd) [reply IndicoV2/Extensions/Jobs/JobAwaiter.cs:30]: `PENDING_AUTOREVIEW` is from the `SubmissionStatus`, and here I'm checking the `JobStatus`. 

Anyway - you were right there are statuses being missing here, I'm changing this part to include `PENDING, RECEIVED, STARTED` as those in progress.

File: IndicoV2.Abstractions/Jobs/IJobsClient.cs [hunk 2]
```diff
@@ -22,5 +22,7 @@ public interface IJobsClient
         /// <param name="jobId"></param>
         /// <returns></returns>
         Task<JToken> GetResultAsync(string jobId);
+
+        Task<string> GetFailReasonAsync(string jobId);
```
[the remark this answers is not in the document]
Comment (rkaczka-vd) [reply IndicoV2.Abstractions/Jobs/IJobsClient.cs]: Changed

File: IndicoV2.V1Adapters/Jobs/V1JobsClientAdapter.cs [hunk 3]
```diff
@@ -20,6 +22,38 @@ public V1JobsClientAdapter(IndicoClient indicoClient, JobStatusConverter jobStat
...
+            };
+            var jobResponse = await _indicoClient.GraphQLHttpClient.SendQueryAsync(query);
+            var job = (JObject)jobResponse.Data.job;
+
+            var jobStatus = (JobStatus)Enum.Parse(typeof(JobStatus), job.Value<string>("status"));
+            
+            if (jobStatus != JobStatus.FAILURE)
+            {
+                throw new InvalidOperationException(
+                    $"Cannot get fail reason, job have not failed (status: {jobStatus}");
```
[the remark this answers is not in the document]
Comment (rkaczka-vd) [reply IndicoV2.V1Adapters/Jobs/V1JobsClientAdapter.cs]: Changed

File: IndicoV2/Extensions/Jobs/JobAwaiter.cs (hunk 1 above)
Comment (goatrocks) [reply IndicoV2/Extensions/Jobs/JobAwaiter.cs]: Ah right, got the enum list confused :grimacing: But thanks for fixing it!

**Review by goatrocks — APPROVED** [review]

**Pull request merged**